H/hyper/build_tarballs.jl

```diff
@@ -3,18 +3,19 @@
 using BinaryBuilder
 
 name = "hyper"
-version = v"0.14.17"
+version = v"0.14.18"
 
 # Collection of sources required to complete build
 sources = [
     ArchiveSource("https://github.com/hyperium/hyper/archive/refs/tags/v$(version).tar.gz",
-                  "64420fd550f43af09b0722b3504d4fd919de642d63f01ad54108aa854f5f5470"),
+                  "6095636d02ea5af3ff5f06d80b9466f7b58ba76339f39813b33c3f24c663fdef"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/hyper*/
-cargo build --release
+sed '21 a [lib]\ncrate-type=["cdylib"]\n' -i Cargo.toml
```

Comment:
Isn't there a way to enable building of the library without having to edit the file manually?  Also, this sed script doesn't seem very sustainable as it hardcodes a line number, probability it'll bit-rot at some point is very high.

Reply:
https://github.com/hyperium/hyper/commit/1c6637060e36654ddb2fdfccb0d146c7ad527476 seems to suggest you can use `--crate-type`?

Reply:
That would require a nightly version of rust. Do we have that available in the environment?

Reply:
No :facepalm: Why do they do this? :weary:

Reply:
For now, we can switch back to v0.14.17 if the sed script appears ugly.

Reply:
I think they made the decision because the C API is unstable, so it is supposed to be used with nightly instead of stable rust.

Reply:
Yeah, let's rollback to 0.14.7 and update once we have a new Rust toolchain with this capability

Reply:
Done!